projects/media-viewer/src/lib/store/reducers/annotatons.reducer.ts

```diff
@@ -168,6 +170,38 @@ export function reducer (
         commentSearchQueries
       };
     }
+
+    case fromAnnotations.APPLY_COMMENT_SUMMARY_FILTER: {
+      const payload = action.payload;
+      const hasTagFilter = () => {
```

Comment:
This feels like it should be in a selector rather than here

Reply:
i prefer it to be here as its clearly visible in the store. when u have business logic in selectors you don't know what is the state of it... only when u need a data from different reducers is when u add business logic in selector as reducers are not async and selectors are